### DISCUSSION TITLE
Bad string generated for Volta NodeJS version

### DISCUSSION THREAD

**Discussion by BryanHuntNV** [renovatebot/renovate#26926] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.157.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Renovate started creating MRs with the following change:

```json

  "volta": {
  -  "node": "18.18.2"
  +  "node": "v20.11.0"
  },
  ```
  
  which causes the error:
  
  ```
Volta error: Could not parse version "v20.11.0"
Please verify the intended version.
Volta error: Error cause: encountered unexpected token: AlphaNumeric("v20")
  ```

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by viceice** [comment]
🙈 @rarkins @secustor it's caused by my node PR.

we should fix node versioning to trim `v` when current value doesn't have it. can be done on getNewValue

**Reply by secustor** [reply]
I would rather add it to the dependency extractions. 
We have added it there in the past and I do not like that we such logic in the versioning if we allow users to set `extractVersion`. It would make this non transparent for users as this wouldn't show up in the logs if I'm not mistaken.

**Reply by viceice** [reply]
- #26918
https://github.com/renovatebot/renovate/blob/9bd0bf1312837568b06c017220790b802b488014/lib/modules/versioning/node/index.spec.ts#L16-L23

it should already trim the leading `v`, so it seems some other versioning is overriding the datasource default 🤔 

https://github.com/renovatebot/renovate/blob/9bd0bf1312837568b06c017220790b802b488014/lib/modules/datasource/node-version/index.ts#L3
https://github.com/renovatebot/renovate/blob/9bd0bf1312837568b06c017220790b802b488014/lib/modules/datasource/node-version/index.ts#L20

**Reply by secustor** [reply]
The npm manager sets it https://github.com/renovatebot/renovate/blob/9bd0bf1312837568b06c017220790b802b488014/lib/modules/manager/npm/index.ts#L20

**Reply by viceice** [reply]
yes, validated https://github.com/renovate-reproductions/26926-node-versioning/pull/6/files

sending a fix pr

**Reply by viceice** [reply, marked as answer]
- #26927
